rpm/0071-sailfishos-webrtc-Implement-video-capture-module.-JB.patch

```diff
@@ -0,0 +1,797 @@
++    const char* deviceUniqueId) {
++    rtc::scoped_refptr<VideoCaptureModuleSFOS> implementation(
++        new rtc::RefCountedObject<VideoCaptureModuleSFOS>());
++
++    if (implementation->Init(deviceUniqueId) != 0)
++        return nullptr;
++
++    return implementation;
++}
++
++VideoCaptureModuleSFOS::VideoCaptureModuleSFOS()
++    : VideoCaptureImpl()
++{
++    mozilla::hal::ScreenConfiguration screenConfig;
++
++    mozilla::hal::RegisterScreenConfigurationObserver (this);
++    mozilla::hal::GetCurrentScreenConfiguration (&screenConfig);
++    _screenRotationAngle = ScreenOrientationToAngle (screenConfig.orientation());
++}
++
++int32_t VideoCaptureModuleSFOS::Init(const char* deviceUniqueIdUTF8)
++{
++    /* Fill current device name for the parent class */
++    int len = strlen(deviceUniqueIdUTF8);
++    _deviceUniqueId = new (std::nothrow) char[len + 1];
++    if (_deviceUniqueId) {
++        memcpy(_deviceUniqueId, deviceUniqueIdUTF8, len + 1);
++    } else {
++        return -1;
++    }
++
++    if (amber_camera_manager()->openCamera(deviceUniqueIdUTF8, _camera)) {
++        amber::media::CameraInfo info;
++        if (_camera->getInfo(info)) {
++            _rearFacingCamera = info.facing == amber::media::AMBER_CAMERA_FACING_REAR;
++            _sensorMountAngle = info.mountAngle;
++            _camera->setListener(this);
++            return 0;
++        }
++        _camera.reset();
++    }
++    return -1;
++}
++
++VideoCaptureModuleSFOS::~VideoCaptureModuleSFOS()
++{
++    mozilla::hal::UnregisterScreenConfigurationObserver(this);
++}
++
++int32_t VideoCaptureModuleSFOS::StartCapture(
++    const VideoCaptureCapability& capability)
++{
++    _startNtpTimeMs = webrtc::Clock::GetRealTimeClock()->CurrentNtpInMilliseconds();
++    UpdateCaptureRotation();
++
++    _requestedCapability = capability;
++    _requestedCapability.codecType = kVideoCodecUnknown;
++    _requestedCapability.rawType = kVideoI420;
++
++    amber::media::CameraCapability cap;
++    cap.width = capability.width;
++    cap.height = capability.height;
++    cap.fps = capability.maxFPS;
++    return _camera->startCapture(cap) ? 0 : -1;
++}
++
++int32_t VideoCaptureModuleSFOS::StopCapture()
++{
++    return _camera->stopCapture() ? 0 : -1;
++}
++
++bool VideoCaptureModuleSFOS::CaptureStarted()
++{
++    return _camera->captureStarted();
++}
++
++void VideoCaptureModuleSFOS::Notify(const mozilla::hal::ScreenConfiguration& aConfiguration)
++{
++    LOG_F(LS_INFO) << "VideoCaptureModuleSFOS::Notify ScreenConfiguration.orientation: " << aConfiguration.orientation();
++    _screenRotationAngle = ScreenOrientationToAngle(aConfiguration.orientation());
++    UpdateCaptureRotation();
++}
++
++int VideoCaptureModuleSFOS::ScreenOrientationToAngle(mozilla::dom::ScreenOrientationInternal orientation)
++{
++    switch (orientation) {
++    /* FIXME: What is the default orientation? */
++    case mozilla::dom::eScreenOrientation_Default:
```

Comment:
Sailfish devices always default to portrait. The SDK emulator might be an exception to that, but that might only be an issue at the compositor level.

Reply:
Thank you. I changed the comment.